contacts.module

```diff
@@ -161,6 +228,52 @@ function template_preprocess_contacts_dash_summary(array &$variables) {
   }
 }
 
+/**
+ * Implements hook_toolbar().
+ */
+function contacts_toolbar() {
+  $items = [];
+  $items['manage_mode'] = [
+    '#cache' => [
+      'contexts' => [
+        'user.permissions',
+      ],
+    ],
+  ];
+
+  if (!\Drupal::currentUser()->hasPermission('administer contacts')) {
+    return $items;
+  }
```

Comment:
Is this 'the right way TM', or do they normally include the items with an `'#access' => FALSE`?

Reply:
This is how core does it in `contextual_toolbar`:

```
function contextual_toolbar() {
  $items = [];
  $items['contextual'] = [
    '#cache' => [
      'contexts' => [
        'user.permissions',
      ],
    ],
  ];

  if (!\Drupal::currentUser()->hasPermission('access contextual links')) {
    return $items;
  }
```

Reply:
Perfect!